lib/src/calendar_week.dart

```diff
@@ -409,54 +414,53 @@ class _CalendarWeekState extends State<CalendarWeek> {
 
   /// Date item layout
   Widget _dateItem(DateTime date) => DateItem(
-        today: _calendarController._today,
-        date: date,
-        dateStyle: compareDate(date, _calendarController._today)
-            ? widget.todayDateStyle
-            : date != null && (date.weekday == 6 || date.weekday == 7)
-                ? widget.weekendsStyle
-                : widget.dateStyle,
-        pressedDateStyle: widget.datePressedStyle,
-        backgroundColor: widget.dateBackgroundColor,
-        todayBackgroundColor: widget.todayBackgroundColor,
-        pressedBackgroundColor: widget.datePressedBackgroundColor,
-        decorationAlignment: () {
-          /// If date is contain in decorations list, use decorations Alignment
-          if (widget.decorations.isNotEmpty) {
-            final List<DecorationItem> matchDate = widget.decorations
-                .where((ele) => compareDate(ele.date, date))
-                .toList();
-            return matchDate.isNotEmpty
-                ? matchDate[0].decorationAlignment
-                : FractionalOffset.center;
-          }
-          return FractionalOffset.center;
-        }(),
-        dayShapeBorder: widget.dayShapeBorder,
-        onDatePressed: (datePressed) {
-          _calendarController._selectedDate = datePressed;
-          widget.onDatePressed(datePressed);
-        },
-        onDateLongPressed: (datePressed) {
-          _calendarController._selectedDate = datePressed;
-          widget.onDateLongPressed(datePressed);
-        },
-        decoration: () {
-          /// If date is contain in decorations list, use decorations Widget
-          if (widget.decorations.isNotEmpty) {
-            final List<DecorationItem> matchDate = widget.decorations
-                .where((ele) => compareDate(ele.date, date))
-                .toList();
-            return matchDate.isNotEmpty ? matchDate[0].decoration : null;
-          }
-          return null;
-        }(),
-        subject: _subject,
-      );
+      today: _calendarController._today,
+      date: date,
+      dateStyle: compareDate(date, _calendarController._today)
```

Comment:
có hàm compare sao không handle logic ở trong hàm luôn. mà lại dùng toán tử nhiều lần thế này. fix đi nhá

Reply:
Ko hỉu comment nói gì cả :v

Reply:
Ý là dùng method này hả 
https://api.dart.dev/stable/2.12.4/dart-core/DateTime/difference.html

Reply:
ý là trong hàm compare đấy sao ko check date xong return widget luôn đi lại còn phải check thêm ở ngoài này làm gì

Reply:
Return `Widget` thì có tái sửa dụng được ko :v. Hay chỉ dùng đc cho view

Reply:
@mduccc thế nên tách ra hàm cho cho dễ nhìn chứ cũng không nên để lồng toán tử thế này

Reply:
Tách ra hàm gì ?
Bản thân `compareDate` là 1 hàm rồi :v

Reply:
Kệ đi :v Cái đó là do MR này
https://github.com/mduccc/flutter_calendar_week/pull/2

Reply:
Improve sau vậy :v

Reply:
Nice!